### ISSUE TITLE
Run rosetta-cli tests against our Rosetta implementation.

### ISSUE THREAD

**Issue by Sventimir** [MinaProtocol/mina#12119]
I'm currently working to run [rosetta-cli ](https://github.com/coinbase/rosetta-cli) against our Rosetta implementation. Rosetta-cli is a test suite for rosetta servers developed by Coinbase and we are using it to verify that our implementation conforms to the specification and standards set by Coinbase (who consumes tha API). I'm trying to run against my local blockchain for the moment, but for a more comprehensive test suite we will probably need to run it against some testnet too.

**Comment by Sventimir** [comment]
While running these tests I discovered some errors in our implementation, that were not specified in the initial scope of work on Rosetta implementation. I'm working to fix those errors as they appear. Problem is that those tests run sequentially, because results of previous tests serve as inputs to later ones, so I cannot know the full extent of failure until I fix all the tests. This is problematic, but for the moment we cannot find a workaround.

**Comment by kantp** [comment]
Moving to blocked, as we're waiting on a reply in the forums/from coinbase.

**Comment by Sventimir** [comment]
This is now blocked by issue #12299 instead.

**Comment by kantp** [comment]
This will be resolved by merging #12317 and #12186.

**Comment by Sventimir** [comment]
It turns out, that the archive is expected to put the genesis ledger into the database at startup, but to do so, it hat to be given the daemon's configuration filename, which I didn't know. However, adding that configuration file to the command starting the archive, in itself does not solve the problem yet. There are other bugs to be solved, which I'm investigating at the moment. #12317 is put on hold and will probaby change significantly before merging.

**Comment by joaosreis** [comment]
`rosetta-cli` tests for the Data API are working and passing successfully on a local setup. They will now be integrated into a CI pipeline.
